static/src/javascripts/projects/common/modules/discussion/activity-stream.js

```diff
@@ -0,0 +1,143 @@
+// @flow
+
+import bonzo from 'bonzo';
+import bean from 'bean';
+import $ from 'lib/$';
+import {
+    getUrlVars,
+    constructQuery,
+    replaceQueryString,
+    pushQueryString,
+} from 'lib/url';
+import Component from 'common/modules/component';
+import discussionApi from 'common/modules/discussion/api';
+
+class PopStateEvent extends Event {
+    state: Object;
+}
+
+const pagination = (activityStream): void => {
+    bean.on(
+        activityStream.elem,
+        'click',
+        '.js-activity-stream-page-change',
+        (e: Event) => {
+            const target: HTMLElement = (e.currentTarget: any);
+            const page: ?string = target.getAttribute('data-page');
+            e.preventDefault();
+
+            activityStream.change({
+                page,
+            });
+        }
+    );
+};
+
+const selectTab = (type: string): void => {
+    // Blur so that when pressing forward/back the focus is not retained on
+    // the old tab Note, without the focus first, the blur doesn't seem to
+    // work for some reason
+    $('.js-activity-stream-change').focus().blur();
+
+    $('.tabs__tab--selected').removeClass('tabs__tab--selected');
+    bonzo($(`a[data-stream-type=${type}]`))
+        .parent()
+        .addClass('tabs__tab--selected');
+};
+
+class ActivityStream extends Component {
+    constructor(opts: Object): void {
+        super();
+
+        this.setOptions(opts);
+        this.endpoint =
+            '/discussion/profile/:userId/:streamType.json?page=:page';
+        this.componentClass = 'activity-stream';
+        this.defaultOptions = {
+            page: 1,
+            streamType: 'discussions',
+            userId: null,
+        };
+    }
+
+    applyState(html: string, streamType: string): void {
+        // update display
+        const $el = bonzo(this.elem).empty();
+
+        this.setState('loading');
+
+        $.create(html).each(el => {
+            $el.html($(el).html()).attr({ class: el.className });
```

Comment:
[Reserved words as property names](http://kangax.github.io/compat-table/es5/#test-Object/array_literal_extensions_Reserved_words_as_property_names) doesn't work in <= IE8, even with ES5 shim.